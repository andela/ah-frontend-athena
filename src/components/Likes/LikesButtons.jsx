import React from "react";
import { MDBContainer, MDBRow, MDBIcon, MDBCol } from "mdbreact";
import PropTypes from "prop-types";
import "./LikeButtons.scss";

const LikesButtons = ({ like, onClick, onClick2, likes_count }) => {
  const icons = like();
  return (
    <div>
      <MDBContainer>
        <MDBRow className="flex">
          <MDBCol md="3" className="offset-md-3">
            <span className={icons.icon1}>{likes_count}</span>
            <MDBIcon
              id="likeBtn"
              icon="thumbs-up"
              size="2x"
              onClick={onClick}
              className={icons.icon1}
            />
            <MDBIcon
              id="unlikeBtn"
              icon="thumbs-down"
              size="2x"
              onClick={onClick2}
              className={icons.icon2}
              flip="horizontal"
            />
          </MDBCol>
        </MDBRow>
      </MDBContainer>
    </div>
  );
};
LikesButtons.propTypes = {
  like: PropTypes.func.isRequired,
  onClick: PropTypes.func.isRequired,
  onClick2: PropTypes.func.isRequired,
  likes_count: PropTypes.string.isRequired
};
export default LikesButtons;
