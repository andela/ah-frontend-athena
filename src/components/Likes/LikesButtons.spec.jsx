import React from "react";
import { shallow } from "enzyme";
import LikesButtons from "./LikesButtons";

let onClick = jest.fn();
let onClick2 = jest.fn();
let like = () => ({ icon1: "", icon2: "" });
let likes_count = "";
let props = {
  onClick,
  onClick2,
  likes_count,
  like
};

describe("Test like buttons", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<LikesButtons {...props} />);
  });
  it("Test if like and dislike buttons mount", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
