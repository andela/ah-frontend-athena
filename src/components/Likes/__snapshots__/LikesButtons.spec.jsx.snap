// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test like buttons Test if like and dislike buttons mount 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LikesButtons
    like={[Function]}
    likes_count=""
    onClick={[MockFunction]}
    onClick2={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <t
        fluid={false}
        tag="div"
      >
        <t
          className="flex"
          tag="div"
        >
          <t
            className="offset-md-3"
            lg={null}
            md="3"
            sm={null}
            tag="div"
            xl={null}
            xs={null}
          >
            <span
              className=""
            >
              
            </span>
            <t
              border={false}
              brand={false}
              className=""
              fab={false}
              fal={false}
              far={false}
              fixed={false}
              flip=""
              icon="thumbs-up"
              id="likeBtn"
              inverse=""
              light={false}
              list={false}
              onClick={[MockFunction]}
              pull=""
              pulse={false}
              regular={false}
              rotate=""
              size="2x"
              spin={false}
              stack=""
            />
            <t
              border={false}
              brand={false}
              className=""
              fab={false}
              fal={false}
              far={false}
              fixed={false}
              flip="horizontal"
              icon="thumbs-down"
              id="unlikeBtn"
              inverse=""
              light={false}
              list={false}
              onClick={[MockFunction]}
              pull=""
              pulse={false}
              regular={false}
              rotate=""
              size="2x"
              spin={false}
              stack=""
            />
          </t>
        </t>
      </t>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <t
          className="flex"
          tag="div"
        >
          <t
            className="offset-md-3"
            lg={null}
            md="3"
            sm={null}
            tag="div"
            xl={null}
            xs={null}
          >
            <span
              className=""
            >
              
            </span>
            <t
              border={false}
              brand={false}
              className=""
              fab={false}
              fal={false}
              far={false}
              fixed={false}
              flip=""
              icon="thumbs-up"
              id="likeBtn"
              inverse=""
              light={false}
              list={false}
              onClick={[MockFunction]}
              pull=""
              pulse={false}
              regular={false}
              rotate=""
              size="2x"
              spin={false}
              stack=""
            />
            <t
              border={false}
              brand={false}
              className=""
              fab={false}
              fal={false}
              far={false}
              fixed={false}
              flip="horizontal"
              icon="thumbs-down"
              id="unlikeBtn"
              inverse=""
              light={false}
              list={false}
              onClick={[MockFunction]}
              pull=""
              pulse={false}
              regular={false}
              rotate=""
              size="2x"
              spin={false}
              stack=""
            />
          </t>
        </t>,
        "fluid": false,
        "tag": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <t
            className="offset-md-3"
            lg={null}
            md="3"
            sm={null}
            tag="div"
            xl={null}
            xs={null}
          >
            <span
              className=""
            >
              
            </span>
            <t
              border={false}
              brand={false}
              className=""
              fab={false}
              fal={false}
              far={false}
              fixed={false}
              flip=""
              icon="thumbs-up"
              id="likeBtn"
              inverse=""
              light={false}
              list={false}
              onClick={[MockFunction]}
              pull=""
              pulse={false}
              regular={false}
              rotate=""
              size="2x"
              spin={false}
              stack=""
            />
            <t
              border={false}
              brand={false}
              className=""
              fab={false}
              fal={false}
              far={false}
              fixed={false}
              flip="horizontal"
              icon="thumbs-down"
              id="unlikeBtn"
              inverse=""
              light={false}
              list={false}
              onClick={[MockFunction]}
              pull=""
              pulse={false}
              regular={false}
              rotate=""
              size="2x"
              spin={false}
              stack=""
            />
          </t>,
          "className": "flex",
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <span
                className=""
              >
                
              </span>,
              <t
                border={false}
                brand={false}
                className=""
                fab={false}
                fal={false}
                far={false}
                fixed={false}
                flip=""
                icon="thumbs-up"
                id="likeBtn"
                inverse=""
                light={false}
                list={false}
                onClick={[MockFunction]}
                pull=""
                pulse={false}
                regular={false}
                rotate=""
                size="2x"
                spin={false}
                stack=""
              />,
              <t
                border={false}
                brand={false}
                className=""
                fab={false}
                fal={false}
                far={false}
                fixed={false}
                flip="horizontal"
                icon="thumbs-down"
                id="unlikeBtn"
                inverse=""
                light={false}
                list={false}
                onClick={[MockFunction]}
                pull=""
                pulse={false}
                regular={false}
                rotate=""
                size="2x"
                spin={false}
                stack=""
              />,
            ],
            "className": "offset-md-3",
            "lg": null,
            "md": "3",
            "sm": null,
            "tag": "div",
            "xl": null,
            "xs": null,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "",
              },
              "ref": null,
              "rendered": "",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "border": false,
                "brand": false,
                "className": "",
                "fab": false,
                "fal": false,
                "far": false,
                "fixed": false,
                "flip": "",
                "icon": "thumbs-up",
                "id": "likeBtn",
                "inverse": "",
                "light": false,
                "list": false,
                "onClick": [MockFunction],
                "pull": "",
                "pulse": false,
                "regular": false,
                "rotate": "",
                "size": "2x",
                "spin": false,
                "stack": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "border": false,
                "brand": false,
                "className": "",
                "fab": false,
                "fal": false,
                "far": false,
                "fixed": false,
                "flip": "horizontal",
                "icon": "thumbs-down",
                "id": "unlikeBtn",
                "inverse": "",
                "light": false,
                "list": false,
                "onClick": [MockFunction],
                "pull": "",
                "pulse": false,
                "regular": false,
                "rotate": "",
                "size": "2x",
                "spin": false,
                "stack": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <t
          fluid={false}
          tag="div"
        >
          <t
            className="flex"
            tag="div"
          >
            <t
              className="offset-md-3"
              lg={null}
              md="3"
              sm={null}
              tag="div"
              xl={null}
              xs={null}
            >
              <span
                className=""
              >
                
              </span>
              <t
                border={false}
                brand={false}
                className=""
                fab={false}
                fal={false}
                far={false}
                fixed={false}
                flip=""
                icon="thumbs-up"
                id="likeBtn"
                inverse=""
                light={false}
                list={false}
                onClick={[MockFunction]}
                pull=""
                pulse={false}
                regular={false}
                rotate=""
                size="2x"
                spin={false}
                stack=""
              />
              <t
                border={false}
                brand={false}
                className=""
                fab={false}
                fal={false}
                far={false}
                fixed={false}
                flip="horizontal"
                icon="thumbs-down"
                id="unlikeBtn"
                inverse=""
                light={false}
                list={false}
                onClick={[MockFunction]}
                pull=""
                pulse={false}
                regular={false}
                rotate=""
                size="2x"
                spin={false}
                stack=""
              />
            </t>
          </t>
        </t>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <t
            className="flex"
            tag="div"
          >
            <t
              className="offset-md-3"
              lg={null}
              md="3"
              sm={null}
              tag="div"
              xl={null}
              xs={null}
            >
              <span
                className=""
              >
                
              </span>
              <t
                border={false}
                brand={false}
                className=""
                fab={false}
                fal={false}
                far={false}
                fixed={false}
                flip=""
                icon="thumbs-up"
                id="likeBtn"
                inverse=""
                light={false}
                list={false}
                onClick={[MockFunction]}
                pull=""
                pulse={false}
                regular={false}
                rotate=""
                size="2x"
                spin={false}
                stack=""
              />
              <t
                border={false}
                brand={false}
                className=""
                fab={false}
                fal={false}
                far={false}
                fixed={false}
                flip="horizontal"
                icon="thumbs-down"
                id="unlikeBtn"
                inverse=""
                light={false}
                list={false}
                onClick={[MockFunction]}
                pull=""
                pulse={false}
                regular={false}
                rotate=""
                size="2x"
                spin={false}
                stack=""
              />
            </t>
          </t>,
          "fluid": false,
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <t
              className="offset-md-3"
              lg={null}
              md="3"
              sm={null}
              tag="div"
              xl={null}
              xs={null}
            >
              <span
                className=""
              >
                
              </span>
              <t
                border={false}
                brand={false}
                className=""
                fab={false}
                fal={false}
                far={false}
                fixed={false}
                flip=""
                icon="thumbs-up"
                id="likeBtn"
                inverse=""
                light={false}
                list={false}
                onClick={[MockFunction]}
                pull=""
                pulse={false}
                regular={false}
                rotate=""
                size="2x"
                spin={false}
                stack=""
              />
              <t
                border={false}
                brand={false}
                className=""
                fab={false}
                fal={false}
                far={false}
                fixed={false}
                flip="horizontal"
                icon="thumbs-down"
                id="unlikeBtn"
                inverse=""
                light={false}
                list={false}
                onClick={[MockFunction]}
                pull=""
                pulse={false}
                regular={false}
                rotate=""
                size="2x"
                spin={false}
                stack=""
              />
            </t>,
            "className": "flex",
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <span
                  className=""
                >
                  
                </span>,
                <t
                  border={false}
                  brand={false}
                  className=""
                  fab={false}
                  fal={false}
                  far={false}
                  fixed={false}
                  flip=""
                  icon="thumbs-up"
                  id="likeBtn"
                  inverse=""
                  light={false}
                  list={false}
                  onClick={[MockFunction]}
                  pull=""
                  pulse={false}
                  regular={false}
                  rotate=""
                  size="2x"
                  spin={false}
                  stack=""
                />,
                <t
                  border={false}
                  brand={false}
                  className=""
                  fab={false}
                  fal={false}
                  far={false}
                  fixed={false}
                  flip="horizontal"
                  icon="thumbs-down"
                  id="unlikeBtn"
                  inverse=""
                  light={false}
                  list={false}
                  onClick={[MockFunction]}
                  pull=""
                  pulse={false}
                  regular={false}
                  rotate=""
                  size="2x"
                  spin={false}
                  stack=""
                />,
              ],
              "className": "offset-md-3",
              "lg": null,
              "md": "3",
              "sm": null,
              "tag": "div",
              "xl": null,
              "xs": null,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "",
                  "className": "",
                },
                "ref": null,
                "rendered": "",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "border": false,
                  "brand": false,
                  "className": "",
                  "fab": false,
                  "fal": false,
                  "far": false,
                  "fixed": false,
                  "flip": "",
                  "icon": "thumbs-up",
                  "id": "likeBtn",
                  "inverse": "",
                  "light": false,
                  "list": false,
                  "onClick": [MockFunction],
                  "pull": "",
                  "pulse": false,
                  "regular": false,
                  "rotate": "",
                  "size": "2x",
                  "spin": false,
                  "stack": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "border": false,
                  "brand": false,
                  "className": "",
                  "fab": false,
                  "fal": false,
                  "far": false,
                  "fixed": false,
                  "flip": "horizontal",
                  "icon": "thumbs-down",
                  "id": "unlikeBtn",
                  "inverse": "",
                  "light": false,
                  "list": false,
                  "onClick": [MockFunction],
                  "pull": "",
                  "pulse": false,
                  "regular": false,
                  "rotate": "",
                  "size": "2x",
                  "spin": false,
                  "stack": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
